src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -143,5 +145,36 @@ internal FileData VisitDictionaryValueNullChecked(string key, FileData node)
 
             return base.VisitFileData(node);
         }
+
+        public override Result VisitResult(Result node)
+        {
+            _ruleId = node.RuleId;
+            node = base.VisitResult(node);
+            _ruleId = null;
+
+            return node;
+        }
+
+        public override Message VisitMessage(Message node)
```

Comment:
>VisitMessage [](start = 32, length = 12)

SARIF string look-up semantics are very complex due to the format supporting localization. We can refer to an external resources file. We have a global strings table. We have per-rule strings tables. And so, this message string flattener will be very useful for pipeline developers. We need to extract this logic into a separate helper, I think, so that viewers can also easily get access to it. I'll do this when I tackle #983 below.